### PR TITLE
Merge ActionStager and CommandAdjuster into PathMapper

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
Stacked on https://github.com/bazelbuild/bazel/pull/17807.

File: src/main/java/com/google/devtools/build/lib/actions/CommandLines.java [hunk 1]
```diff
@@ -110,27 +110,27 @@ private CommandLines(Object commandLines) {
    *
    * @param artifactExpander The artifact expander to use.
    * @param paramFileBasePath Used to derive param file names. Often the first output of an action
-   * @param pathStripper function to strip configuration prefixes from output paths, in accordance
+   * @param pathMapper function to strip configuration prefixes from output paths, in accordance
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/actions/CommandLines.java:113]: sanity checking again: "strip" is still the right word in the comment?

File: src/main/java/com/google/devtools/build/lib/actions/PathStripper.java [hunk 2]
```diff
@@ -72,182 +74,139 @@
  */
 public final class PathStripper {
   /**
-   * Support for stripping config paths from an action's inputs and outputs.
+   * Creates a new {@link PathMapper} for action implementation logic to use.
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/actions/PathStripper.java:77]: I'm a bit confused by the wording at the end: "action implementation logic to use". Especially where right under `class PathStripper` it mentions `creates a PathMapper`. 

Are you intending further refactoring here? If so, no worries. If not I'd try to be more precise.

**Review by gregestren — COMMENTED** [review]
Were you planning further interface merging? Or is this PR the final step?

File: src/main/java/com/google/devtools/build/lib/actions/PathStripper.java (hunk 2 above)
Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/actions/PathStripper.java:77]: Yes, I would propose a further change as a follow-up that would clean up this comment: Make `PathMapper` a regular top-level class defined in a separate file and entangle the particular implementations created by `PathStripper` from it.

**Review by gregestren — APPROVED** [review]

**Comment by sgowroji** [comment]
Hi @fmeum, Could you please confirm whether the PR is ready to import as it is still drafted. Thanks!

**Comment by fmeum** [comment]
@sgowroji Yes, ready to be imported without changes. It's stacked on #17807, so that would need to be merged first.

**Comment by aranguyen** [comment]
@sgowroji the first pr https://github.com/bazelbuild/bazel/pull/17807 has been merge, please go ahead with the import.

**Comment by sgowroji** [comment]
Hi @fmeum, Since it's currently awaiting merge, could you please resolve the above code conflicts. Thanks!

**Comment by fmeum** [comment]
@sgowroji I resolved the conflicts.

File: src/main/java/com/google/devtools/build/lib/actions/PathStripper.java [hunk 3]
```diff
@@ -72,58 +74,110 @@
...
-    ActionStager NOOP = execPath -> execPath;
+  /**
+   * Creates a new {@link PathMapper} for executor implementation logic to use.
+   *
+   * @param spawn      the action to stage. If {@link Spawn#stripOutputPaths()} is true, paths like
+   *                   "bazel-out/k8-fastbuild/bin/foo" are reduced to "bazel-out/bin/foo". Else
+   *                   they're unchanged.
+   * @param outputRoot the root path where outputs are written (e.g. "bazel-out")
+   */
+  public static PathMapper createForExecutor(Spawn spawn, PathFragment outputRoot) {
```
Comment (aranguyen) [inline src/main/java/com/google/devtools/build/lib/actions/PathStripper.java:104]: @fmeum small proposal. What do you think about naming these methods `createPathMapperForExecutor` and `createPathMapperForAction` for readability? You mentioned in one of your previous comments to Greg's review that there will be further refactoring of PathMapper so maybe that will address the distinction I'm trying to achieve with renaming here?

**Pull request closed without merging**